### PR TITLE
feat(server): reimplement bundling, watcher, script runner

### PR DESCRIPTION
**Pull request by StarpTech**
- reimplement bundling, watcher and script execution. Move logic to different modules.
- Track uncaught Node.js via https://nodejs.org/api/process.html#event-uncaughtexceptionmonitor
- Graceful shutdown hooks server 

#### Checklist

- [x] run `make test`
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: cli/commands/start.go [hunk 1]
```diff
@@ -9,10 +9,10 @@ import (
 	"path/filepath"
 	"time"
 
+	cmd2 "github.com/go-cmd/cmd"
```
Comment (jensneuse) [inline cli/commands/start.go:12]: maybe `gocmd`?

File: cli/commands/start.go [hunk 2]
```diff
@@ -48,7 +48,12 @@ If used without --exclude-server, make sure the server is available in this dire
 			return err
 		}
 
-		var hooksBundler *bundleconfig.Bundler
+		quit := make(chan os.Signal, 1)
+		signal.Notify(quit, os.Interrupt)
```
Comment (jensneuse) [inline cli/commands/start.go:52]: Should we add `syscall.SIGTERM` as well?
https://stackoverflow.com/a/16538055

File: cli/commands/start.go [hunk 3]
```diff
@@ -60,27 +65,48 @@ If used without --exclude-server, make sure the server is available in this dire
...
+				fmt.Sprintf("HOOKS_TOKEN=%s", hooksJWT),
+				fmt.Sprintf("WG_MIDDLEWARE_PORT=%d", middlewareListenPort),
+				fmt.Sprintf("WG_LISTEN_ADDR=%s", listenAddr),
+			)
+
+			hooksServerDone = make(chan struct{})
+
+			go func() {
+				defer close(hooksServerDone)
+				// Done when both channels have been closed
```
Comment (jensneuse) [inline cli/commands/start.go:88]: are you sure?

File: cli/commands/start.go (hunk 1 above)
Comment (StarpTech) [reply cli/commands/start.go:12]: Yes :+1:

File: cli/commands/start.go (hunk 2 above)
Comment (StarpTech) [reply cli/commands/start.go:52]: Will reiterate on this, thanks.

File: cli/commands/start.go (hunk 3 above)
Comment (StarpTech) [reply cli/commands/start.go:88]: Works and sounds reasonable with the attached references. Code from https://github.com/go-cmd/cmd/blob/master/examples/blocking-streaming/main.go#L28

File: cli/commands/start.go (hunk 2 above)
Comment (StarpTech) [reply cli/commands/start.go:52]: Added

**Pull request merged**